javascript/api.ts

```diff
@@ -0,0 +1,17 @@
+
+export interface IConn {
+	read(len: number): Promise<Uint8Array|undefined>;
+	write(buffer: Uint8Array): Promise<number>;
+	close(): Promise<void>;
+}
+
+export interface ISession {
+	open(): Promise<IChannel>;
+	accept(): Promise<IChannel>;
+    close(): Promise<void>;
+
+}
```

Comment:
The indentation here is inconsistent, looks like a mix of tabs and spaces. I'm seeing that in the other files too.

Reply:
ugh, kill me. i will see if there is a formatter plugin for vscode

Reply:
Yeah, I'm not super-familiar with the formatting options myself. IIRC we were using `TSLint` at work, but that's kind of deprecated now. The VSCode built-in TypeScript language support does seem to offer formatting though, so I ran through converting tabs to spaces and running the formatter with this PR: #7.

Reply:
I also had Prettier installed from some previous stuff, but figured sticking with the built-in one is simpler if it works well enough. It changes some of the inline spacing around operators, and removed some of the alignment of fields. There may be options for some of those things, but I haven't explored that yet.

Reply:
👍